tests/test_carrier_account.py

```diff
@@ -93,3 +93,41 @@ def test_carrier_account_create_with_custom_workflow(prod_client):
             [error["field"] == "account_number" and error["message"] == "must be present and a string"]
             for error in error.errors
         )
+
+
+@pytest.mark.vcr()
+def test_carrier_account_create_ups(prod_client):
+    """Test registering a UPS Carrier Account.
+
+    We purposefully don't pass data here because real data is required for this endpoint
+    which we don't have in a test context, simply assert the error matches when no data is passed.
+    """
+    carrier_account = {
+        "type": "UpsAccount",
+    }
+
+    try:
+        prod_client.carrier_account.create_ups(**carrier_account)
+    except ApiError as error:
+        assert error.http_status == 422
+        assert any(
+            [error["field"] == "account_number" and error["message"] == "must be present and a string"]
+            for error in error.errors
+        )
+
+
+@pytest.mark.vcr()
+def test_carrier_account_update_ups(prod_client):
+    """Test updating a UPS Carrier Account.
+
+    We purposefully don't pass data here because real data is required for this endpoint
+    which we don't have in a test context, simply assert that we routed the request correctly.
+    """
+    carrier_account = {
+        "type": "UpsAccount",
+    }
+
+    try:
+        prod_client.carrier_account.update_ups("ca_123", **carrier_account)
+    except ApiError as error:
+        assert error.http_status == 404
```

Comment:
I just revisited this after finishing the Java library. We should try to avoid asserting a 404 error here because it might not be hitting the correct endpoint. Instead, you can reuse the test from the update above and simply pass in bad data to get a 422 error from the UPS update endpoint. This will give us more confidence that we are actually hitting the correct endpoint, as we can't truly test this.

Reply:
Actually double checking this, the reason I had it as 404 is we can't create a valid UPS account so there wouldn't be one to update. That's why the best we can do is 404 which isn't great but then we can look at the cassettes to make sure it's hitting the right place. Mind taking a look again?

Reply:
You don't need to create a UPS account. Instead, you can reuse the [carrier create](https://github.com/EasyPost/easypost-python/blob/master/tests/test_carrier_account.py#L19) and then pass the carrier ID into the UPS create endpoint. This approach will result in a 422 error, which is much better than receiving a 404. Without looking at the cassette, it would be difficult for future readers to determine whether the 404 occurred due to an incorrect URL endpoint or an invalid carrier ID.

Reply:
This only works today because they haven't yet turned off the old method of creating a UPS account. In a couple weeks, this will stop working so we should not rely on it.

Reply:
We are creating `DhlEcsAccount`: https://github.com/EasyPost/examples/blob/master/official/fixtures/client-library-fixtures.json#L37